### PR TITLE
fix(docs): dead links; add markdown link check

### PR DESCRIPTION
**Pull request by alexeldeib**
digitaloceans trial site worked manually in my browser, but it didn't seem to like the crawler. Maybe a useragent issue of some kind?

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -21,5 +21,9 @@ jobs:
           node-version: '14'
       - name: Install markdownlint
         run: npm install -g markdownlint-cli
+      - name: Install markdownlint
+        run: npm install -g markdown-link-check
       - name: markdownlint
         run: just lint-docs
+      - name: markdown-link-check
+        run: just lint-docs
```
Comment (bacongobbler) [inline .github/workflows/lint.yml:29]: did you mean `just linkcheck-docs`?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -21,5 +21,9 @@ jobs:
           node-version: '14'
       - name: Install markdownlint
         run: npm install -g markdownlint-cli
+      - name: Install markdownlint
```
Comment (bacongobbler) [inline .github/workflows/lint.yml:24]: did you mean "Install markdown-link-check"?

**Review by bacongobbler — CHANGES_REQUESTED** [review]
saw a few typos. Overall looks good though!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (alexeldeib) [reply .github/workflows/lint.yml:29]: whoops -- I swear I had this right locally 🤔

Comment (alexeldeib) [reply .github/workflows/lint.yml]: that's better I think?

**Comment by thomastaylor312** [comment]
@alexeldeib This good for a re-review?

**Comment by alexeldeib** [comment]
yep! just rebased + checked locally, think this is still good.

File: docs/howto/README.md [hunk 3]
```diff
@@ -14,8 +14,6 @@ quickly accomplish common tasks.
 - [Running Krustlet on any Kubernetes cluster with
   inlets](krustlet-with-inlets.md)
 - [Running Krustlet on MicroK8s](krustlet-on-microk8s.md)
-- [Running Kubernetes on Amazon Elastic Kubernetes Service
```
Comment (thomastaylor312) [inline docs/howto/README.md:17]: Did you mean to remove this?

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Just one thing that looks like it may be an accidental removal?

File: docs/howto/README.md (hunk 3 above)
Comment (alexeldeib) [reply docs/howto/README.md:17]: yes -- it's a dupe of the 2nd line with a different name :) 

not picking on AWS, I promise!!

Comment (alexeldeib) [reply docs/howto/README.md:17]: actually, just realized there's a dupe of Kind too

Comment (thomastaylor312) [reply docs/howto/README.md:17]: Ok sweet!

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**